FinniversKit/Sources/Components/SideScrollableView/SideScrollableView.swift

```diff
@@ -0,0 +1,188 @@
+        return CGSize(width: UIView.noIntrinsicMetric,
+                      height: height)
+    }
+
+    public weak var delegate: SideScrollableViewDelegate?
+    
+    // MARK: - Init
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+    
+    // MARK: - Setup
+    
+    public func configure(with viewModel: SideScrollableItemModel) {
+        cleanup()
+        createButtons(withTitles: viewModel.items)
+        subscribeToButtonTaps()
+    }
+
+    private func setup() {
+        addSubview(scrollView)
+        scrollView.addSubview(indicatorView)
+
+        // Bind scrollView to the view boundaries
+        NSLayoutConstraint.activate([
+            scrollView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            scrollView.topAnchor.constraint(equalTo: topAnchor),
+            scrollView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            scrollView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+        
+        // Bind the stackView to the scroll view's content layout guide to
+        // set the content size.
+        NSLayoutConstraint.activate([
+            contentView.leadingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.leadingAnchor),
+            contentView.topAnchor.constraint(equalTo: scrollView.contentLayoutGuide.topAnchor),
+            contentView.trailingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.trailingAnchor),
+            contentView.bottomAnchor.constraint(equalTo: scrollView.contentLayoutGuide.bottomAnchor)
+        ])
+        
+        indicatorViewLeadingConstraint = indicatorView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor)
+        indicatorViewWidthConstraint = indicatorView.widthAnchor.constraint(equalToConstant: 0)
+        
+        NSLayoutConstraint.activate([
+            indicatorViewLeadingConstraint,
+            indicatorView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor),
+            indicatorViewWidthConstraint
+        ])
+    }
+
+    private func cleanup() {
+        cancellables.removeAll()
+        buttons.removeAll()
+        contentView.removeArrangedSubviews()
+    }
+    
+    private func createButtons(withTitles itemTitles: [String]) {
+        // The current implementation always makes the first item as selected
+        for (index, itemName) in itemTitles.enumerated() {
+            let button = Button.makeSideScrollableButton(withTitle: itemName)
+            contentView.addArrangedSubview(button)
+            buttons.append(ButtonItem(title: itemName,
+                                      button: button))
+
+            button
+                .publisher(for: .touchUpInside)
+                .sink { [weak self, index] _ in
+                    self?.selectedItemPublisher.send(index)
```

Comment:
I think I would argue that we actually don't need this `selectedItemPublisher` as it's not published out of the view. Also we could reduce the number of subscriptions by one. 

My proposal is to simply remove it, and instead rename `subscribeToButtonTap` to `handleButtonTap(_ index: Int)` and call it directly from this sink. That way we also remove the need for weak assigning in `subscribeToButtonTap`.

```
button
  .publisher(for: .touchUpInside)
  .sink { [weak self, index] _ in 
    self?.handleButtonTap(index, button)
  }
  .store(in: &cancellables)
``` 

and

```
private func handleButtonTap(_ index: Int, _ button: UIButton) {
  let selectedItem = buttons[selectedIndex]
  scrollView.scrollRectToVisible(button.frame, animated: true)

  indicatorViewWidthConstraint.constant = button.frame.width
  indicatorViewLeadingConstraint.constant = button.frame.
  
  UIView.animate(withDuration: 0.2, delay: 0, options: .curveEaseOut) {
    self?.layoutIfNeeded()
  }

  delegate?.sidescrollableViewDidTapItem(self, item: selectedItem.title, itemIndex: index)
}
```

Reply:
That's great, thanks 🙌 Somehow I felt that the second publisher is kind of unnecessary 😄